financialaid/serializers.py

```diff
@@ -61,3 +83,75 @@ def save(self):
         # Add auditing here
 
         return financial_aid
+
+
+class FinancialAidActionSerializer(serializers.Serializer):
+    """
+    Serializer for financial aid actions
+    """
+    action = ChoiceField(
+        choices=[
+            FinancialAidStatus.REJECTED,
+            FinancialAidStatus.APPROVED,
+            FinancialAidStatus.PENDING_MANUAL_APPROVAL
+        ],
+        write_only=True
+    )
+    tier_program_id = IntegerField(write_only=True)
+
+    def validate(self, data):
+        """
+        Validators for this serializer
+        """
+        # Check that the previous financial aid status allows for the new status
+        if (data['action'] == FinancialAidStatus.REJECTED
+                and self.instance.status != FinancialAidStatus.PENDING_MANUAL_APPROVAL):
+            raise ValidationError("Cannot reject application that is not pending manual approval.")
+        if (data['action'] == FinancialAidStatus.APPROVED
+                and self.instance.status != FinancialAidStatus.PENDING_MANUAL_APPROVAL):
+            raise ValidationError("Cannot approve application that is not pending manual approval.")
+        if (data['action'] == FinancialAidStatus.PENDING_MANUAL_APPROVAL
+                and self.instance.status != FinancialAidStatus.PENDING_DOCS):
+            raise ValidationError("Cannot mark documents as received for application not pending docs.")
+        # Check tier program exists
+        try:
+            data["tier_program"] = TierProgram.objects.get(
+                id=data["tier_program_id"],
+                program=self.instance.tier_program.program_id
```

Comment:
Wouldn't this have a slight performance decrease since it'll be hitting the database again to fetch `.program` instead of just using `.program_id` on `tier_program`?